Doc/whatsnew/3.14.rst

```diff
@@ -189,6 +189,11 @@ asyncio
 Deprecated
 ==========
 
+* Using :meth:`~object.__class_getitem__` on a metaclass
```

Comment:
"Using `__class_getitem__` on a metaclass" doesn't seem like the right description of what we're deprecating. It's fine to have a metaclass that implements `__class_getitem__`, as long as it is used only on subclasses of the metaclass, not instances of it.

The situation where this deprecation applies is complex and a bit hard to describe. I would suggest something like: "Currently, subscripting a class object with a metaclass that defines the `__class_getitem__` method would invoke the metaclass's method. This behavior is now deprecated and will be removed in Python 3.16. To make instances of a metaclass subscriptable, define a `__getitem__` method on the metaclass."